src/login/logind-button.c

```diff
@@ -181,6 +219,19 @@ static int button_dispatch(sd_event_source *s, int fd, uint32_t revents, void *u
                         break;
                 }
 
+        } else if (ev.type == EV_KEY && ev.value == 0) {
+                if (ev.code == KEY_RESTART) {
+                        log_struct(LOG_INFO,
+                                   LOG_MESSAGE("Reboot key released."),
+                                   "MESSAGE_ID=" SD_MESSAGE_REBOOT_KEY_RELEASE_STR);
+
+                        if (b->manager->reboot_key_longpress_event_source) {
+                                b->manager->reboot_key_longpress_event_source = sd_event_source_unref(b->manager->reboot_key_longpress_event_source);
+
+                                manager_handle_action(b->manager, INHIBIT_HANDLE_REBOOT_KEY, b->manager->handle_reboot_key, b->manager->reboot_key_ignore_inhibited, true);
+                        }
+                }
```

Comment:
hmm, is this the right approach?

i.e. should the actin be triggered 5s after button down? or shouldn't the action be triggered once button is up again? The latter sounds a lot more natural to me?

Reply:
My reasoning is:
1. Users do not need to count seconds, guessing if they waited enough time to trigger long press.
2. It's consistent with the behaviour of the ATX power button for making forced poweroff. User don't need to release button to trigger atx poweroff.

Anyway, I will leave the final word on this at your side. I can implement it either way.

Reply:
hmm, ok, i guess going by time since button down makes sense indeed.